src/Featurizers/ForecastingPivotFeaturizer.h

```diff
@@ -0,0 +1,129 @@
+///  \class         ForecastingPivotTransformer
+///  \brief         Expand output from upstream featurizers where all values
+///                 in the same column across matrixes are not null/empty
+///
+template <typename InputT>
+class ForecastingPivotTransformer :
+    public Components::InferenceOnlyTransformerImpl<
+        typename std::vector<Microsoft::Featurizer::RowMajMatrix<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>>,
+        std::vector<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>
+    > {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType =
+        Components::InferenceOnlyTransformerImpl<
+            typename std::vector<Microsoft::Featurizer::RowMajMatrix<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>>,
+            std::vector<typename Microsoft::Featurizer::Traits<InputT>::nullable_type>
+        >;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    ForecastingPivotTransformer(void) = default;
+    ForecastingPivotTransformer(Archive &ar);
+
+    ~ForecastingPivotTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(ForecastingPivotTransformer);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Types
+    // |
+    // ----------------------------------------------------------------------
+    using NullableType                      = typename Microsoft::Featurizer::Traits<InputT>::nullable_type;
+    using MatrixType                        = Microsoft::Featurizer::RowMajMatrix<NullableType>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+
+    // MSVC runs into problems when the declaration and definition are separated
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+        if (input.empty()) {
+            return;
+        }
+        
+        std::vector<typename Microsoft::Featurizer::Traits<InputT>::nullable_type> ret;
+        Eigen::Index const col_size(input[0].cols());
+
+        for (Eigen::Index col = 0; col < col_size; ++col) {
+            bool has_nan(false);
+
+            // if any nans is found in the current column across all matrixes, this column is omitted
+            for (MatrixType const & matrix : input) {
+                // all matrixes must have the same column number
+                assert(matrix.cols() == col_size);
```

Comment:
>assert [](start = 16, length = 6)

This should be an actual error rather than an assert (which is only active for debug builds).

Reply:
Raising exceptions for empty input and matrixes with different column number now. Unit tests also added for these two cases.